napalm/base/test/getters.py

```diff
@@ -40,19 +40,36 @@ def dict_diff(prv, nxt):
     result = {}
 
     for k in keys:
-        if isinstance(prv.get(k), dict):
-            if isinstance(nxt.get(k), dict):
-                "If both are dicts we do a recursive call."
-                diff = dict_diff(prv.get(k), nxt.get(k))
+        a = prv.get(k)
+        b = nxt.get(k)
+
+        if isinstance(a, dict):
+            if isinstance(b, dict):
+                # If both are dicts we do a recursive call.
+                diff = dict_diff(a, b)
+                if diff:
+                    result[k] = diff
+            else:
+                # If only one is a dict they are clearly different
+                result[k] = {"result": a, "expected": b}
+        elif isinstance(a, list) and isinstance(b, list):
+            if any(isinstance(item, dict) for item in a) or any(
+                isinstance(item, dict) for item in b
+            ):
+                # Compare the lists of dictionaries
+                diff = list_dicts_diff(a, b)
```

Comment:
Related to line 68 in the same file. See some comment below.

One of the nice side effects is that it shows the specific key that doesn’t match, instead of the entire dictionary.

Reply:
This was reverted.

It might be worth looking into it though. It made comparing failed results so much easier...